rules/otp.bzl

```diff
@@ -4,163 +4,50 @@
-                # Set the alert timeout cycles to 0 per class.
-                "OWNER_SW_CFG_ROM_ALERT_TIMEOUT_CYCLES": ["0x00000000"] * 4,
-                # Set the alert phase cycles to 0,10,10,0xFFFFFFFF for classes
-                # A and B, and to all zeros for classes C and D.
-                "OWNER_SW_CFG_ROM_ALERT_PHASE_CYCLES": [
-                    "0x0",
-                    "0xa",
-                    "0xa",
-                    "0xFFFFFFFF",
-                    "0x0",
-                    "0xa",
-                    "0xa",
-                    "0xFFFFFFFF",
-                    "0x0",
-                    "0x0",
-                    "0x0",
-                    "0x0",
-                    "0x0",
-                    "0x0",
-                    "0x0",
-                    "0x0",
-                ],
-                "OWNER_SW_CFG_ROM_ALERT_DIGEST_RMA": "0x36ed9cb0",
-            },
-        },
-        {
-            "name": "HW_CFG",
-            # If set to true, this computes the HW digest value and locks the
-            # partition.
-            "lock": True,
-            "items": {
-                "DEVICE_ID": "<random>",
-                # Cryptolib and chip-level tests require access to the CSRNG
-                # software interfaces.
-                "EN_CSRNG_SW_APP_READ": True,
-                # Cryptolib and chip-level tests require access to the
-                # entropy_src FW data interface.
-                "EN_ENTROPY_SRC_FW_READ": True,
-                # Cryptolib and chip-level tests require access to the
-                # entropy_src FW override interface.
-                "EN_ENTROPY_SRC_FW_OVER": True,
-            },
-        },
-        {
-            "name": "SECRET0",
-            "lock": True,
-            "items": {
-                "TEST_UNLOCK_TOKEN": "<random>",
-                "TEST_EXIT_TOKEN": "<random>",
-            },
-        },
-        {
-            "name": "SECRET1",
-            "lock": True,
-            "items": {
-                "FLASH_ADDR_KEY_SEED": "<random>",
-                "FLASH_DATA_KEY_SEED": "<random>",
-                "SRAM_DATA_KEY_SEED": "<random>",
-            },
-        },
-        {
-            "name": "SECRET2",
-            "lock": False,
-            "items": {
-                "RMA_TOKEN": "<random>",
-                "CREATOR_ROOT_KEY_SHARE0": "<random>",
-                "CREATOR_ROOT_KEY_SHARE1": "<random>",
-            },
-        },
-        {
-            "name": "LIFE_CYCLE",
-            "state": ctx.attr.lc_state,
-            # Can range from 0 to 16.
-            # Note that a value of 0 is only permissible in RAW state.
-            "count": ctx.attr.lc_count,
-        },
-    ]
+def otp_partition(name, **kwargs):
+    partition = {
+        "name": name,
+    }
+    partition.update(kwargs)
+    return json.encode(partition)
+
+def _otp_json_impl(ctx):
+    otp = {}
+    if ctx.attr.seed != "":
+        otp["seed"] = ctx.attr.seed
```

Comment:
A quick heads-up about seeds: 

Right now, the DV environment will also need the ability to optionally pass the `--lc-seed` and `--otp-seed` arguments to the generator script. 

I should note though that after M2 we will have to replace the RNG mechanism used in these scripts by the entropy buffer file mechanism explained in https://github.com/lowRISC/opentitan/issues/15144 (see doc linked on that issue). 

For simulation / emulation purposes, reseeding the design and OTP image then essentially becomes a two-step process where a separate script will have to be called to generate new buffer files prior to the SW and HW builds (topgen will have to be rerun as well). For the tapeout on the other hand, the plan is to use an HSM in the secure domain to create high quality entropy buffer files.

The seed arguments should then likely become file paths pointing to individual entropy buffer files (they can be handled by BAZEL dependency management as well). We can discuss this more once this initial PR has landed, though.